keydata.go

```diff
@@ -512,6 +594,26 @@ func (d *KeyData) updatePassphrase(payload, oldAuthKey []byte, passphrase string
 	return nil
 }
 
+func (d *KeyData) updatePIN(oldAuthKey []byte, pin PIN, platformContext any) error {
+	handlerInfo, exists := keyDataHandlers[d.data.PlatformName]
+	if !exists {
+		return ErrNoPlatformHandlerRegistered
+	}
+
+	authKey, err := d.derivePINAuthKey(pin)
+	if err != nil {
+		return err
+	}
+
+	handle, err := handlerInfo.handler.ChangeAuthKey(d.platformKeyData(), oldAuthKey, authKey, platformContext)
+	if err != nil {
+		return err
+	}
+
+	d.data.PlatformHandle = handle
+	return nil
+}
+
 func (d *KeyData) openWithPassphrase(passphrase string) (payload []byte, authKey []byte, err error) {
```

Comment:
there is no openWithPIN because there is no extra encryption with PINs, right?

Reply:
That's right - the PIN derived key is only used for authentication.